### DISCUSSION TITLE
Map over a list of kwargs dictionaries

### DISCUSSION THREAD

**Discussion by multimeric** [PrefectHQ/prefect#5014] [Q&A]
Say I have one function that creates an iterable of keyword arguments (ie a dictionary), and I want to then map this using another task. For example:
```python
import prefect
import typing

@prefect.task
def generate_kwargs() -> typing.Iterable[dict]:
    for i in range(10):
        yield dict(a=i, b=i**2, c=i**3)

@prefect.task
def uses_kwargs(a, b, c):
    return a + b + c

with prefect.Flow("Run Me") as flow:
    kwargs = generate_kwargs()
    uses_kwargs.map(kwargs)    
```
Now of course this returns:
```
TypeError: missing a required argument: 'b'
```

However is there any way to actually achieve this? Because there is no point at which I can intercept the `map` to tell prefect to use `**kwargs` anywhere.

**Comment by leonarduschen** [comment]
Is there a way to achieve this yet?

**Reply by neumann-nico** [reply]
In prefect 2.0 you can use a for loop instead of map. I am not sure if yield should be used in tasks, but running locally this worked:
```python
import typing

from prefect import flow, get_run_logger, task


@task
def generate_kwargs() -> typing.Iterable[dict]:
    for i in range(10):
        yield dict(a=i, b=i**2, c=i**3)


@task
def uses_kwargs(a, b, c):
    return a + b + c


@flow(name="Run Me")
def flow_run():
    logger = get_run_logger()
    for kwargs in generate_kwargs():
        output = uses_kwargs(**kwargs)
        logger.info(output)


flow_run()
```

**Reply by pbierkortte** [reply]
@multimeric,  get_run_logger() is expecting a kw argument 'b', this has nothing to do with Prefect. Also, except for pandas, list.map() is not common, and most of the time list comprehensions are clearer. Lastly, if you really want to use that list map pattern, you can do it like list(map(function(), iterable)).
